gpu_ocean/prototypes/scripts/mpi_run_benchmark.py

```diff
@@ -0,0 +1,313 @@
+# -*- coding: utf-8 -*-
+
+"""
+This software is part of GPU Ocean. 
+
+Copyright (C) 2018 SINTEF Digital
+Copyright (C) 2018 Norwegian Meteorological Institute
+
+This python program runs a short program while reporting execution
+metrics, and is intended to be used for meassuring computational 
+performance and throughput.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import sys, os
+from mpi4py import MPI
+current_dir = os.path.dirname(os.path.realpath(__file__))
+
+
+if os.path.isdir(os.path.abspath(os.path.join(current_dir, '../../SWESimulators'))):
+        sys.path.insert(0, os.path.abspath(os.path.join(current_dir, '../../')))
+
+import argparse
+parser = argparse.ArgumentParser(description='Benchmark a simulator.')
+parser.add_argument('--nx', type=int, default=2048)
+parser.add_argument('--ny', type=int, default=2048)
+parser.add_argument('--block_width', type=int)
+parser.add_argument('--block_height', type=int)
+parser.add_argument('--steps_per_download', type=int, default=2000)
+parser.add_argument('--iterations', type=int, default=1)
+parser.add_argument('--simulator', type=str)
+parser.add_argument('--output', type=str, default=None)
+
+args = parser.parse_args()
+
+
+# Import timing utilities
+import time
+tic = time.time();
+
+# Import packages we need
+import numpy as np
+import json
+from SWESimulators import FBL, CTCS, KP07, CDKLM16, PlotHelper, Common
+
+
+toc = time.time()
+print("{:02.4f} s: ".format(toc-tic) + "Imported packages")
+
+# Create CUDA context
+tic = time.time()
+gpu_ctx = Common.CUDAContext()
+device_name = gpu_ctx.cuda_device.name()
+toc = time.time()
+print("{:02.4f} s: ".format(toc-tic) + "Created context on " + device_name)
+
+# Set benchmark sizes
+dx = 200.0
+dy = 200.0
+
+dt = 0.95/100
+g = 9.81
+
+f = 0.00
+r = 0.0
+
+boundaryConditions = Common.BoundaryConditions()
+
+# Generate initial conditions
+waterHeight = 60
+
+x_center = dx*args.nx/2.0
+y_center = dy*args.ny/2.0
+size = 0.4*min(args.nx*dx, args.ny*dy)
+
+def my_exp(i, j):
+        x = dx*i - x_center
+        y = dy*j - y_center
+        return np.exp(-10*(x*x/(size*size)+y*y/(size*size)))
+        
+"""
+Initializes the KP simulator
+"""
+def initKP():
```

Comment:
I Think we can skip all except CDKLM for now

Reply:
(y)